chartfx-chart/src/main/java/de/gsi/chart/legend/spi/DefaultLegend.java

```diff
@@ -106,9 +107,17 @@ protected double computePrefWidth(final double forHeight) {
 
     public LegendItem getNewLegendItem(final Renderer renderer, final DataSet series, final int seriesIndex) {
         final Canvas symbol = renderer.drawLegendSymbol(series, seriesIndex, SYMBOL_WIDTH, SYMBOL_HEIGHT);
-        return new LegendItem(series.getName(), symbol);
+        var item = new LegendItem(series.getName(), symbol);
+        if(renderer.supportsVisibility()){
```

Comment:
I wonder whether we want this optional or as a default requirement. This is a nice feature and not too much to ask for renderer implementers to include. 

Alternatively -- if implementers forget to add/implement this -- the data set wouldn't be taken into the auto-range calculation but still been drawn. Als an interesting effect but not sure whether this is really bad or actually good....

Reply:
If it is only possible to toggle to non-visible if the renderer supports it, I think it should be safe to do it this way. So i guess it really boils down to whether supporting this extra logic is more effort than just implementing it for all renderers.

Reply:
You're probably right. I made it a default requirement and added it to all renderers I could find.